test/test_helper.rb

```diff
@@ -0,0 +1,145 @@
+  'Content-Type' => 'application/json'
+}
+
+def index_response(resource, count)
+  {
+    :total_count => count,
+    :has_more => false,
+    :object => 'list',
+    :data => FactoryGirl.build_list(resource.to_sym, count)
+  }.to_json
+end
+
+def resource_from_uri(uri)
+  uri.to_s.split('/').last.singularize.to_sym
+end
+
+# configure test-unit for FactoryGirl
+class Minitest::Test
+  include WebMock::API
+  include FactoryGirl::Syntax::Methods
+
+  def setup
+    stub_request(:any, /.*api\.blockscore\.com\/.*/).
+      with(headers: HEADERS).
+      to_return do |request|
+        match = URL_REGEXP.match request.uri.to_s
+        factory_name = resource_from_uri(match[:resource])
+
+        # id and action might be "" (empty string)
+        id = (match[:id].nil? || match[:id].empty?) ? nil : match[:id]
+        action = (match[:action].nil? || match[:action].empty?) ? nil : match[:action]
+
+        factory = FactoryGirl.factories[factory_name]
+        if factory.nil?
+          raise ArgumentError, "could not find factory #{factory_name.inspect}."
+        end
+
+        status = 200 # will only change on CREATEs
+        if id == nil && request.method == :get
+          # index
+          body = index_response(factory_name, 5)
+        elsif action == 'history'
+          body = FactoryGirl.build_list(factory_name, 5).to_json
+        elsif action == 'hits'
+          body = index_response(factory_name, 5)
+        else
+          status = 201 if request.method == :post && action.nil?
+          body = FactoryGirl.json(factory_name)
+        end
+
+        { :status => status, :body => body, :headers => {} }
+      end
+  end
+end
+
+# Convert a resource into the corresponding BlockScore class.
+def resource_to_class(resource)
+  "BlockScore::#{resource.camelcase}".constantize
+end
+
+module ResourceTest
+  def self.included(base)
+    base.mattr_accessor :resource
+    base.resource = base.to_s[/^(\w+)ResourceTest/, 1].underscore
+  end
+
+  def test_create_resource
+    response = TestClient.send(:"create_#{resource}")
+    assert_equal response.class, resource_to_class(resource)
+  end
+
+  def test_retrieve_resource
+    r = TestClient.send(:"create_#{resource}")
+    response = resource_to_class(resource).send(:retrieve, r.id)
+    assert_equal resource, response.object
+  end
+
+  def test_list_resource
+    response = resource_to_class(resource).send(:all)
+    assert_equal Array, response.class
+  end
+
+  def test_list_resource_with_count
+    msg = "List #{resource} with count = 2 failed"
+    response = resource_to_class(resource).send(:all, {:count => 2})
+    assert_equal Array, response.class, msg
+  end
+
+  def test_list_resource_with_count_and_offset
+    msg = "List #{resource} with count = 2 and offset = 2 failed"
+    response = resource_to_class(resource).
+      send(:all, {:count => 2, :offset => 2})
+    assert_equal Array, response.class, msg
+  end
+end
+
+class TestClient
```

Comment:
I'm not sure I understand the point of `TestClient`.  When I look at the tests I just feel like I don't understand what is really being tested.